api/server.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+// Defines missing consts in the API Spec
+const (
+	ApplicationvndVeraisonCharesJson string = "application/vnd.veraison.chares+json"
+)
+
+type Server struct {
+	logger *zap.SugaredLogger
+}
+
+func NewServer(logger *zap.SugaredLogger) *Server {
+	return &Server{
+		logger: logger,
+	}
+}
+
+func (s *Server) returnBadRequest(w http.ResponseWriter, r *http.Request, errMsg string) {
+	s.logger.Error(errMsg)
+	badRequestError := &BadRequestError{
+		Detail: &errMsg,
+		Status: N400,
+		Title:  InvalidRequest,
+		Type:   TagGithubCom2024VeraisonratsdErrorInvalidrequest,
+	}
+	w.WriteHeader(http.StatusBadRequest)
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(badRequestError)
+}
+
+func (s *Server) RatsdChares(w http.ResponseWriter, r *http.Request, param RatsdCharesParams) {
+	var requestData ChaResRequest
+
+	// Check if content type matches the expectation
+	ct := r.Header.Get("Content-Type")
+	if ct != ApplicationvndVeraisonCharesJson {
+		errMsg := fmt.Sprintf("wrong content type, expect %s (got %s)", ApplicationvndVeraisonCharesJson, ct)
+		s.returnBadRequest(w, r, errMsg)
+		return
+	}
+
+	respCt := fmt.Sprintf(`application/eat+jwt; eat_profile=%q`, TagGithubCom2024Veraisonratsd)
+	if *(param.Accept) != respCt {
+		errMsg := fmt.Sprintf("wrong accept type, expect %s (got %s)", respCt, *(param.Accept))
+		w.WriteHeader(http.StatusNotAcceptable)
+		w.Write([]byte(errMsg))
```

Comment:
This error (like any other error) should be packaged into a RFC9457 "Problem Details" payload.

We don't need to fix it now.  I'll raise an issue to track this.